rust_sample/ci/build_sample.py

```diff
@@ -73,15 +73,15 @@ def post_function(config, packages):
         "packages": {
             "rust_sample": {
                 "example_binary": "example_binary",
-                "rust_sample": "librust_sample.a",
+                "rust_sample": "librust_sample.dylib",
```

Comment:
Similarly, why do we change the type of the artifact in this PR?

Reply:
I've encountered a version mismatch when linking and this fixes it, also we're using dynamic linking on libtelio from now on.

Reply:
I'm a little bit confused - what we do in libtelio should have no impact on the sample project in this repo. Is this change needed only after updating to latest rust or was simply it not noticed before?

Reply:
After updating to latest rust the linker was also updated so I had to change the iOS deployment version which affected also the sample project here. For some reason I couldn't understand why the static lib was still using the old version of the linker, so I've changed this to the dynamic libs where the new linker is used.